pkg/deploy/lattice/access_log_subscription_manager_test.go

```diff
@@ -438,6 +435,200 @@ func TestAccessLogSubscriptionManager(t *testing.T) {
+				SourceName:        sourceName,
+				DestinationArn:    s3DestinationArn,
+				ALPNamespacedName: accessLogPolicyNamespacedName,
+				EventType:         core.UpdateEvent,
+			},
+			Status: &lattice.AccessLogSubscriptionStatus{
+				Arn: accessLogSubscriptionArn,
+			},
+		}
+		updateALSInput := &vpclattice.UpdateAccessLogSubscriptionInput{
+			AccessLogSubscriptionIdentifier: aws.String(accessLogSubscriptionArn),
+			DestinationArn:                  aws.String(s3DestinationArn),
+		}
+		updateALSErr := &vpclattice.ConflictException{
+			ResourceType: aws.String("ACCESS_LOG_SUBSCRIPTION"),
+		}
+		serviceNetworkInfo := &services.ServiceNetworkInfo{
+			SvcNetwork: vpclattice.ServiceNetworkSummary{
+				Arn:  aws.String(serviceNetworkArn),
+				Name: aws.String(sourceName),
+			},
+		}
+		createALSInput := &vpclattice.CreateAccessLogSubscriptionInput{
+			ResourceIdentifier: aws.String(serviceNetworkArn),
+			DestinationArn:     aws.String(s3DestinationArn),
+			Tags:               expectedTags,
+		}
+		createALSOutput := &vpclattice.CreateAccessLogSubscriptionOutput{
+			Arn: aws.String(newAccessLogSubscriptionArn),
+		}
+		deleteALSInput := &vpclattice.DeleteAccessLogSubscriptionInput{
+			AccessLogSubscriptionIdentifier: aws.String(accessLogSubscriptionArn),
+		}
+		deleteALSOutput := &vpclattice.DeleteAccessLogSubscriptionOutput{}
+
+		mockLattice.EXPECT().UpdateAccessLogSubscriptionWithContext(ctx, updateALSInput).Return(nil, updateALSErr)
+		mockLattice.EXPECT().FindServiceNetwork(ctx, sourceName, config.AccountID).Return(serviceNetworkInfo, nil)
+		mockLattice.EXPECT().CreateAccessLogSubscriptionWithContext(ctx, createALSInput).Return(createALSOutput, nil)
+		mockLattice.EXPECT().DeleteAccessLogSubscriptionWithContext(ctx, deleteALSInput).Return(deleteALSOutput, nil)
+
+		mgr := NewAccessLogSubscriptionManager(gwlog.FallbackLogger, cloud)
+		resp, err := mgr.Update(ctx, accessLogSubscription)
+		assert.Nil(t, err)
+		assert.Equal(t, newAccessLogSubscriptionArn, resp.Arn)
+	})
+
+	t.Run("Update_ALSDoesNotExist_ReturnsInvalidError", func(t *testing.T) {
+		accessLogSubscription := &lattice.AccessLogSubscription{
+			Spec: lattice.AccessLogSubscriptionSpec{
+				SourceType:        lattice.ServiceNetworkSourceType,
+				SourceName:        sourceName,
+				DestinationArn:    s3DestinationArn,
+				ALPNamespacedName: accessLogPolicyNamespacedName,
+				EventType:         core.UpdateEvent,
+			},
+			Status: &lattice.AccessLogSubscriptionStatus{
+				Arn: accessLogSubscriptionArn,
+			},
+		}
+		updateALSInput := &vpclattice.UpdateAccessLogSubscriptionInput{
+			AccessLogSubscriptionIdentifier: aws.String(accessLogSubscriptionArn),
+			DestinationArn:                  aws.String(s3DestinationArn),
+		}
+		updateALSError := &vpclattice.ResourceNotFoundException{
+			ResourceType: aws.String("ACCESS_LOG_SUBSCRIPTION"),
+		}
+
+		mockLattice.EXPECT().UpdateAccessLogSubscriptionWithContext(ctx, updateALSInput).Return(nil, updateALSError)
```

Comment:
Just a thought, which can reduce code for unit tests, makes it easier to read and change in future. 

In this mock we do 2 things, 1. assert input into lattice client is correct and 2. it produces output which use later in manager.

And every test here asserts that input to lattice is correct. I dont think we need that, we can create a single test that makes sure we call Lattice with expected fields populated. For remaining tests we use any() in place of input. For example:

```go
		mockLattice.EXPECT().
			FindService(gomock.Any(), gomock.Any()).
			Return(&vpclattice.ServiceSummary{
				Arn:  aws.String("svc-arn"),
				Id:   aws.String("svc-id"),
				Name: aws.String(svc.LatticeServiceName()),
			}, nil)
```

Reply:
it was a minor comment, I'm ok with current version

Reply:
I think I prefer the more explicit nature of the existing code